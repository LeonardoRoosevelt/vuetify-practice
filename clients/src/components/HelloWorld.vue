<template>
  <v-container>
    <v-row class="text-center">
      <v-col cols="12">
        <v-img
          :src="require('../assets/logo.svg')"
          class="my-3"
          contain
          height="200"
        />
      </v-col>

      <v-col class="mb-4">
        <h1 class="display-2 font-weight-bold">
          {{ title }}
        </h1>
      </v-col>
      <v-col class="mb-4">
        <v-btn class="display-2 font-weight-bold" data-testid="button" @click="title = 'Hello'">
        </v-btn>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
export default {
  name: 'HelloWorld',

  data: () => ({
    title: 'Welcome to Vuetify Practice'
  })
}
</script>
