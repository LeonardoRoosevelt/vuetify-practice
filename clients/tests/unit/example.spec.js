import { createLocalVue, mount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'
import Vuetify from 'vuetify';

describe('HelloWorld.vue', () => {
  let localVue
  let vuetify
  let wrapper
  beforeEach(() => {
    localVue = createLocalVue(); // because of vuetify, we should use a localVue instance
    vuetify = new Vuetify();
    wrapper = mount(HelloWorld, {
      localVue,
      vuetify
    });
  });
  test("True Test", () => {
    const h1 = wrapper.get('h1')
    expect(h1.text()).toBe('Welcome to Vuetify Practice')
  });
  test("Button Test", async () => {
    const button = wrapper.find('[data-testid="button"]')
    expect(button.exists()).toBe(true)
    button.vm.$emit('click')
    await wrapper.vm.$nextTick()
    const h1 = wrapper.get('h1')
    expect(h1.text()).toBe('Hello')
  });
})