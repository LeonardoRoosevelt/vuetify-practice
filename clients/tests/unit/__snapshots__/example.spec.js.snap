// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue True Test 1`] = `
<div class="container" title="Welcome to Vuetify Practice">
  <div class="row text-center">
    <div class="col col-12">
      <div class="v-image v-responsive my-3 theme--light" style="height: 200px;">
        <div class="v-responsive__content"></div>
      </div>
    </div>
    <div class="mb-4 col">
      <h1 class="display-2 font-weight-bold">
        Welcome to Vuetify Practice
      </h1>
    </div>
  </div>
</div>
`;
